thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/pojo/SearchFilters.java

```diff
@@ -0,0 +1,228 @@
+package com.linkedin.thirdeye.dashboard.resources.v2.pojo;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.thirdeye.anomalydetection.context.AnomalyFeedback;
+import com.linkedin.thirdeye.api.DimensionMap;
+import com.linkedin.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+
+public class SearchFilters {
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private static final Logger LOG = LoggerFactory.getLogger(SearchFilters.class);
+
+  Map<String, Integer> statusFilterMap;
+
+  Map<String, Integer> functionFilterMap;
+
+  Map<String, Integer> datasetFilterMap;
+
+  Map<String, Integer> metricFilterMap;
+
+  Map<String, Map<String, Integer>> dimensionFilterMap;
+
+  public SearchFilters() {
+
+  }
+
+  public SearchFilters(Map<String, Integer> statusFilterMap, Map<String, Integer> functionFilterMap, Map<String, Integer> datasetFilterMap,
+      Map<String, Integer> metricFilterMap, Map<String, Map<String, Integer>> dimensionFilterMap) {
+    super();
+    this.statusFilterMap = sortByValue(statusFilterMap, true);
+    this.functionFilterMap = sortByValue(functionFilterMap, true);
+    this.datasetFilterMap = sortByValue(datasetFilterMap, true);
+    this.metricFilterMap = sortByValue(metricFilterMap, true);
+    this.dimensionFilterMap = new TreeMap<>();
+    for (String dimensionName : dimensionFilterMap.keySet()) {
+      this.dimensionFilterMap.put(dimensionName, sortByValue(dimensionFilterMap.get(dimensionName), true));
+    }
+  }
+
+  public static <K, V extends Comparable<? super V>> Map<K, V> sortByValue(Map<K, V> map) {
+    return sortByValue(map, false);
+  }
+
+  public static <K, V extends Comparable<? super V>> Map<K, V> sortByValue(Map<K, V> map, final boolean descending) {
+    final int sortMultipler = (descending) ? -1 : 1;
+    List<Map.Entry<K, V>> list = new LinkedList<Map.Entry<K, V>>(map.entrySet());
```

Comment:
We could use arraylist here since we are not going to insert or remove elements in/from the middle of the list.